docs/architecture/microservices/architect-microservice-container-applications/microservice-based-composite-ui-shape-layout.md

```diff
@@ -19,11 +19,11 @@ At client application start-up time, each of the client UI components (TypeScrip
 
 Figure 4-21 shows a version of this composite UI approach. This is simplified because you might have other microservices that are aggregating granular parts that are based on different techniques. It depends on whether you're building a traditional web approach (ASP.NET MVC) or an SPA (Single Page Application).
 
-![In composite UI application, each UI section is generated by a UI composition microservice, that acts like a mini-gateway.](./media/image21.png)
+![Diagram of a composite UI made up of many view models.](./media/microservice-based-composite-ui-shape-layout/microservice-generate-composite-ui.png)
 
 **Figure 4-21**. Example of a composite UI application shaped by back-end microservices
 
-Each of those UI composition microservices would be similar to a small API Gateway. But in this case each one is responsible for a small UI area.
+In composite UI application, each UI section is generated by a UI composition microservice, that acts like a mini-gateway. Each of those UI composition microservices would be similar to a small API Gateway. But in this case each one is responsible for a small UI area.
```

Comment:
It seems this would repeat information

```suggestion
Each of those UI composition microservices would be similar to a small API Gateway. But in this case, each one is responsible for a small UI area.
```